Sources/TorusUtils/Extensions/TorusUtils+extension.swift

```
@@ -110,108 +117,105 @@ extension TorusUtils {
         var rpcdata : Data = Data.init()
         do {
             if let loadedStrings = try NSKeyedUnarchiver.unarchiveTopLevelObjectWithData(extraParams) as? [String:Any] {
-                let newValue = ["verifieridentifier":verifier, "verifier_id": verifierId, "nodesignatures": nodeSignatures, "idtoken": tokenCommitment] as [String:AnyObject]
-                let keepingCurrent = loadedStrings.merging(newValue) { (current, _) in current }
-                
-                // todo : look into hetrogeneous array encoding
+                let value = ["verifieridentifier":verifier, "verifier_id": verifierId, "nodesignatures": nodeSignatures, "idtoken": tokenCommitment] as [String : Any]
+                let keepingCurrent = loadedStrings.merging(value) { (current, _) in current }
+                // TODO : Look into hetrogeneous array encoding
                 let dataForRequest = ["jsonrpc": "2.0",
                                       "id":10,
                                       "method": "ShareRequest",
                                       "params": ["encrypted": "yes",
                                                  "item": [keepingCurrent]]] as [String : Any]
-                rpcdata = try! JSONSerialization.data(withJSONObject: dataForRequest)
+                rpcdata = try JSONSerialization.data(withJSONObject: dataForRequest)
             }
         } catch {
-            self.logger.error("RetrieveIndividualNodeShare: Couldn't read file.")
+            self.logger.error("retrieveDecryptAndReconstruct - error:", error)
```

this will swallow the error. throw from here